### PR TITLE
Add more EC2 instance costs

### PR DESCRIPTION
**Pull request by aliscott**
**Objective**:

Add more options for `aws_instance`.

**Example output**:

```
  aws_instance.t3_unlimited
  ├─ CPU credits                                                                  0  vCPU-hours   0.0500       0.0000        0.0000
  ├─ Linux/UNIX usage (on-demand, t3.medium)                                    730  hours        0.0416       0.0416       30.3680
  └─ Storage (root_block_device)                                                  8  GB-months    0.1000       0.0011        0.8000
  Total                                                                                                        0.0427       31.1680

  aws_instance.ebs_optimized
  ├─ EBS-optimized usage                                                        730  hours        0.0200       0.0200       14.6000
  ├─ Linux/UNIX usage (on-demand, r3.xlarge)                                    730  hours        0.3330       0.3330      243.0900
  └─ Storage (root_block_device)                                                  8  GB-months    0.1000       0.0011        0.8000
  Total                                                                                                        0.3541      258.4900
```

**Pricing details**:

 * EBS-optimized usage costs are charged for previous generation instances that have `ebs_optimized = true` set.
 * t3 and t4g instances launch by default in "Unlimited" mode which means that CPU credit costs can apply. This depends on the number of vCPU-hours that are used by these instances over the baseline. t2 instances also support "Unlimited" mode, but this is disabled by default.

**Status**:

- [x] Add EBS-optimized costs 
- [x] Add CPU credits prices
- [x] Add a warning if tenancy is host
- [x] Update "Compute" to "Linux/UNIX Usage" to be consistent with AWS pricing page
- [x] Add integration tests

**Others changes**:

- [x] Refactor logging
- [x] Add provider cache to speed up tests

**Useful links**:

- Pricing: https://aws.amazon.com/ec2/pricing/on-demand/
- Terraform: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/instance

### CODE REVIEW

File: cmd/infracost/main.go [hunk 1]
```diff
@@ -62,8 +53,8 @@ func main() {
 			},
 			&cli.StringFlag{
 				Name:  "log-level",
-				Usage: "Log level (TRACE, DEBUG, INFO, WARN, ERROR)",
-				Value: "WARN",
+				Usage: "Log level (trace, debug, info, warn, error, fatal)",
```
Comment (alikhajeh1) [inline cmd/infracost/main.go:56]: can you possibly update `cmd/infracost-generate-docs/main.go` so it's consistent?

File: internal/providers/terraform/aws/instance.go [hunk 2]
```diff
@@ -35,27 +42,75 @@ func computeCostComponent(d *schema.ResourceData, region string, purchaseOption
...
-			ProductFamily: strPtr("Compute Instance"),
-			AttributeFilters: []*schema.AttributeFilter{
-				{Key: "instanceType", Value: strPtr(instanceType)},
-				{Key: "tenancy", Value: strPtr(tenancy)},
-				{Key: "operatingSystem", Value: strPtr("Linux")},
-				{Key: "preInstalledSw", Value: strPtr("NA")},
-				{Key: "capacitystatus", Value: strPtr("Used")},
+	costComponents := []*schema.CostComponent{
+		{
+			Name:           fmt.Sprintf("Linux/UNIX Usage (%s, %s)", purchaseOptionLabel, instanceType),
```
Comment (alikhajeh1) [inline internal/providers/terraform/aws/instance.go:47]: Linux/UNIX Usage -> Linux/UNIX usage ?

See `upper/lower case` in https://github.com/infracost/infracost/blob/master/CONTRIBUTING.md, we can change that if we like though.

File: internal/providers/terraform/aws/instance.go [hunk 3]
```diff
@@ -35,27 +42,75 @@ func computeCostComponent(d *schema.ResourceData, region string, purchaseOption
...
 			},
 		},
-		PriceFilter: &schema.PriceFilter{
-			PurchaseOption: &purchaseOption,
-		},
 	}
+
+	if d.Get("ebs_optimized").Bool() {
+		costComponents = append(costComponents, &schema.CostComponent{
+			Name:                 "EBS-Optimized Usage",
```
Comment (alikhajeh1) [inline internal/providers/terraform/aws/instance.go:71]: Usage -> usage?

File: internal/providers/terraform/aws/instance.go [hunk 4]
```diff
@@ -35,27 +42,75 @@ func computeCostComponent(d *schema.ResourceData, region string, purchaseOption
...
+	cpuCredits := d.Get("credit_specification.0.cpu_credits").String()
+	if cpuCredits == "" && (strings.HasPrefix(instanceType, "t3.") || strings.HasPrefix(instanceType, "t4g.")) {
+		cpuCredits = "unlimited"
+	}
+
+	if cpuCredits == "unlimited" {
+		prefix := strings.SplitN(instanceType, ".", 2)[0]
+
+		costComponents = append(costComponents, &schema.CostComponent{
+			Name:           "CPU Credits",
```
Comment (alikhajeh1) [inline internal/providers/terraform/aws/instance.go:97]: CPU credits?

**Review by alikhajeh1 — APPROVED** [review]
LGTM

File: cmd/infracost/main.go (hunk 1 above)
Comment (aliscott) [reply cmd/infracost/main.go:56]: :+1: will fix this and other resources to be consistent

File: internal/providers/terraform/aws/instance.go (hunk 2 above)
Comment (aliscott) [reply internal/providers/terraform/aws/instance.go]: 👍 will fix this and other resources to be consistent

**Pull request merged**